corehq/toggles.py

```diff
@@ -127,22 +127,44 @@ def all_toggles():
     """
     Loads all toggles
     """
+    return all_toggles_by_name_in_scope(globals()).values()
+
+
+def all_toggles_by_name():
     # trick for listing the attributes of the current module.
     # http://stackoverflow.com/a/990450/8207
-    for toggle_name, toggle in globals().items():
+    return all_toggles_by_name_in_scope(globals())
+
+
+def all_toggles_by_name_in_scope(scope_dict):
+    result = {}
+    for toggle_name, toggle in scope_dict.items():
         if not toggle_name.startswith('__'):
             if isinstance(toggle, StaticToggle):
-                yield toggle
+                result[toggle_name] = toggle
+    return result
 
 
 def toggles_dict(username=None, domain=None):
     """
     Loads all toggles into a dictionary for use in JS
+
+    (only enabled toggles are included)
     """
     return {t.slug: True for t in all_toggles() if (t.enabled(username) or
                                                     t.enabled(domain))}
 
 
+def toggle_values_by_name(username=None, domain=None):
+    """
+    Loads all toggles into a dictionary for use in JS
+
+    all toggles (including those not enabled) are included
+    """
+    return {toggle_name: (toggle.enabled(username) or toggle.enabled(domain))
+            for toggle_name, toggle in all_toggles_by_name().items()}
```

Comment:
Toggles come with some built-in caching, so I think this is fine. If the particular kind of caching it does turns out to be insufficient, there's a pretty easy caching solution to this particular problem. My attitude here is to write the code the way you'd want to write it, and then if performance is a problem fix the way the interface is cached. In this particular case I believe the caching is sufficient, and also the amount data and the frequency with which it'll be requested are both low enough that I'm not too worried in the short term.